### PR TITLE
CORE-230 Update jni wrapper compilation

### PR DESCRIPTION
**Pull request by ig-garcia**

### CODE REVIEW

**Comment by ig-garcia** [comment]
Main changes are:
- Compiling sequoia backend and libetpan directly from source instead of copying them for each architecture sources
- Update the default path for yml2 so that compilation can be done easily by default without providing custom .conf files.
This collaborates with EFA-453 so the compilation is made easily in the default case.

**Comment by Stereodark** [comment]
> Compiling sequoia backend and libetpan directly from source instead of copying them for each architecture sources

Why do we make this change? If needed it makes sense, but it consumes more resources, doesn't it?

> Update the default path for yml2 so that compilation can be done easily by default without providing custom .conf files.
> This collaborates with EFA-453 so the compilation is made easily in the default case.

This is very good

File: android/external/Makefile.conf [hunk 1]
```diff
@@ -44,11 +44,6 @@ BOTAN_VERSION=2.19.3
 # Crypto backend used
 CRYPTO_LIB_NAME ?= botan2
 
-### Git deps repos
-EXTERNAL_GIT_REPOS += libetpan|git@github.com:plancksecurity/foundation-libetpan.git?v3.3.17
-# TODO PEMA-103 move this to not external anymore, it comes with the engine
```
Comment (Stereodark) [inline android/external/Makefile.conf:49]: was this story driving the internalization of libetpan?

**Comment by saschabacardit** [comment]
Note that all the fdt adjacent deps have been appended with foundation-; this applies to the JNI wrapper and more, and might cause issues (albeit a rename fixes this)

**Review by saschabacardit — APPROVED** [review]

**Comment by ig-garcia** [comment]
> > Compiling sequoia backend and libetpan directly from source instead of copying them for each architecture sources
> 
> Why do we make this change? If needed it makes sense, but it consumes more resources, doesn't it?

Does it consume more resources? I did not know that. The main reason is taking the dependencies out of the jni wrapper. So we can provide the sources for sequoia backend and libetpan from wherever we want. Like for example as submodules of the containing project.

File: android/external/Makefile.conf (hunk 1 above)
Comment (ig-garcia) [reply android/external/Makefile.conf:49]: Well it is just that having these versions in code, we always need to keep up them to date, which does not seem to make sense if we have a system of submodules. Anyway I can move that to a later ticket if it does not fit well in this one.

**Comment by Stereodark** [comment]
> > > Compiling sequoia backend and libetpan directly from source instead of copying them for each architecture sources
> > 
> > 
> > Why do we make this change? If needed it makes sense, but it consumes more resources, doesn't it?
> 
> Does it consume more resources? I did not know that. The main reason is taking the dependencies out of the jni wrapper. So we can provide the sources for sequoia backend and libetpan from wherever we want. Like for example as submodules of the containing project.

ok makes sense. The consumption of resources I refer to is just that we need to compile them all the time and I don't think we should, especially for stuff that we do not change much.

**Comment by ig-garcia** [comment]
> > > > Compiling sequoia backend and libetpan directly from source instead of copying them for each architecture sources
> > > 
> > > 
> > > Why do we make this change? If needed it makes sense, but it consumes more resources, doesn't it?
> > 
> > 
> > Does it consume more resources? I did not know that. The main reason is taking the dependencies out of the jni wrapper. So we can provide the sources for sequoia backend and libetpan from wherever we want. Like for example as submodules of the containing project.
> 
> ok makes sense. The consumption of resources I refer to is just that we need to compile them all the time and I don't think we should, especially for stuff that we do not change much.

No, of course we should not be compiling them all the time, it should just be once

**Pull request merged**